instance/utils.py

```diff
@@ -218,3 +219,44 @@ def choices(cls):
     def names(cls):
         """Return enum as list of string names """
         return list(prop.name for prop in cls)
+
+
+def create_new_deployment(
+        instance,
+        creator=None,
+        deployment_type=None,
+        cancel_pending=False,
+        **kwargs,
+):
+    """
+    Create a new deployment for an existing instance, and start it asynchronously
+    """
+    # pylint: disable=cyclic-import, useless-suppression
+    from instance.models.deployment import DeploymentType
+    from instance.models.openedx_deployment import DeploymentState, OpenEdXDeployment
+    from instance.tasks import start_deployment
+    from userprofile.models import UserProfile
+
+    changes = None
+    if instance.betatestapplication_set.exists():
+        changes = build_instance_config_diff(instance.betatestapplication_set.get())
+
+    deployment_type = deployment_type or DeploymentType.unknown
+    creator_profile = creator and UserProfile.objects.get(user_id=creator)
+    if cancel_pending:
+        deployment = instance.get_latest_deployment()
+        if deployment.status() in (DeploymentState.changes_pending, DeploymentState.provisioning):
+            deployment.cancel_deployment()
+    deployment = OpenEdXDeployment.objects.create(
+        instance_id=instance.ref.id,
+        creator=creator_profile,
+        type=deployment_type,
+        changes=changes,
+    )
+    # We schedule a deployment instead of starting right now, which gives user a few mins to update the instance
+    # without us having to setup and teardown the appserver.
```

Comment:
Can we add some logging here so that the log is visible under the instance "Log" tab in the "old" manage interface?